bcda/cclf/csv_test.go

```diff
@@ -0,0 +1,273 @@
+			err := s.importer.ImportCSV(test.filepath)
+			if test.err == nil {
+				assert.Nil(s.T(), err)
+			} else {
+				assert.NotNil(s.T(), err)
+				assert.Contains(s.T(), err.Error(), test.err.Error())
+			}
+			cclfRecords := postgrestest.GetCCLFFilesByName(tt, s.db, filepath.Clean(test.filepath))
+			if len(cclfRecords) != 0 {
+				assert.Equal(tt, 1, len(cclfRecords))
+				assert.Equal(tt, filename, cclfRecords[0].Name)
+				beneRecords, _ := postgrestest.GetCCLFBeneficiaries(s.db, int(cclfRecords[0].ID))
+				assert.Equal(s.T(), len(test.cclfBeneRec), len(beneRecords))
+				for i, v := range beneRecords {
+					fmt.Println(i, v)
+					assert.Contains(s.T(), test.cclfBeneRec, (strings.ReplaceAll(v, " ", "")))
+				}
+			} else {
+				assert.Equal(tt, 0, len(cclfRecords))
+			}
+		})
+	}
+
+}
+
+func (s *CSVTestSuite) TestProcessCSV_Integration() {
+
+	file := csvFile{
+		metadata: csvFileMetadata{
+			name:         "P.PCPB.M2411.D191005.T0209260",
+			env:          "test",
+			acoID:        "FOOACO",
+			cclfNum:      8,
+			perfYear:     24,
+			timestamp:    time.Now(),
+			deliveryDate: time.Now(),
+			fileID:       0,
+			fileType:     1,
+		},
+		data: bytes.NewReader([]byte("MBIS\nMBI000001\nMBI000002\nMBI000003")),
+	}
+
+	expectedFile := models.CCLFFile{
+		Name:            "P.PCPB.M2411.D191005.T0209260",
+		ACOCMSID:        "FOOACO",
+		PerformanceYear: 24,
+	}
+	tests := []struct {
+		name       string
+		file       csvFile
+		fileRecord models.CCLFFile
+		mbiRecord  []string
+		err        error
+	}{
+		{"Import CSV attribution success", file, expectedFile, []string{"MBI000001", "MBI000002", "MBI000003"}, nil},
+		{"Import CSV attribution that already exists", file, models.CCLFFile{}, []string{}, errors.New("already exists")},
+	}
+
+	for _, test := range tests {
+		s.T().Run(test.name, func(tt *testing.T) {
+			err := s.importer.ProcessCSV(file)
+			if test.err != nil {
+				err := s.importer.ProcessCSV(file)
+				assert.NotNil(s.T(), err)
+				assert.Contains(s.T(), err.Error(), test.err.Error())
+			} else {
+				assert.Nil(tt, err)
+				cclfRecord := postgrestest.GetCCLFFilesByName(s.T(), s.db, file.metadata.name)
+				assert.Equal(s.T(), 1, len(cclfRecord))
+				assert.Equal(s.T(), expectedFile.Name, cclfRecord[0].Name)
+				assert.Equal(s.T(), expectedFile.ACOCMSID, cclfRecord[0].ACOCMSID)
+				assert.Equal(s.T(), expectedFile.PerformanceYear, cclfRecord[0].PerformanceYear)
+				assert.Equal(s.T(), constants.ImportComplete, cclfRecord[0].ImportStatus)
+
+				beneRecords, _ := postgres.NewRepository(s.db).GetCCLFBeneficiaryMBIs(context.Background(), cclfRecord[0].ID)
+				sort.Strings(beneRecords)
+				assert.Equal(s.T(), 3, len(beneRecords))
+				for i, v := range beneRecords {
+					fmt.Println(i, v)
+					assert.Contains(s.T(), []string{"MBI000001", "MBI000002", "MBI000003"}, (strings.ReplaceAll(v, " ", "")))
+				}
+			}
+
+		})
+	}
+}
+
+// func (s *CSVTestSuite) TestProcessCSV() {
```

Comment:
Still walking through tests to find any cases I may have missed; leaving this here for now but will remove or cleanup before final review.